### PR TITLE
Add methods for removing file from registry and clearing it

### PR DESCRIPTION
**Pull request by MAXakaWIZARD**
Fixes issue #328

### CODE REVIEW

File: src/Gaufrette/Filesystem.php [hunk 1]
```diff
@@ -344,4 +344,22 @@ private function isFileInRegister($key)
     {
         return array_key_exists($key, $this->fileRegister);
     }
+
+    /**
+     * Clear files register
+     */
+    public function clearCache() {
```
Comment (rande) [inline src/Gaufrette/Filesystem.php]: you need to respect CS

Comment (MAXakaWIZARD) [inline src/Gaufrette/Filesystem.php]: @rande thanks! My bad, missed those curly braces))

**Comment by MAXakaWIZARD** [comment]
@wysow Any thoughts about this PR?

**Comment by wysow** [comment]
@MAXakaWIZARD sounds great thanks! Can you add the tests for this methods please?

File: src/Gaufrette/Filesystem.php [hunk 2]
```diff
@@ -344,4 +344,24 @@ private function isFileInRegister($key)
     {
         return array_key_exists($key, $this->fileRegister);
     }
+
+    /**
+     * Clear files register
+     */
+    public function clearCache()
```
Comment (gquemener) [inline src/Gaufrette/Filesystem.php]: I would name it `clear()`

Comment (MAXakaWIZARD) [inline src/Gaufrette/Filesystem.php]: I think `clearCache` is more precice, cause this is cache actually. @wysow what do you think?

**Comment by MAXakaWIZARD** [comment]
@wysow please review

File: src/Gaufrette/Filesystem.php (hunk 2 above)
Comment (wysow) [inline src/Gaufrette/Filesystem.php]: Maybe `clearFileRegister` as it's clearly what it does...

**Comment by wysow** [comment]
@MAXakaWIZARD Thx again, the tests are red, can you take a look and fix them please?

**Comment by MAXakaWIZARD** [comment]
@wysow Tests failed due to github problems, I will perform refactoring and push once more

**Comment by MAXakaWIZARD** [comment]
@wysow So tests are green, method names are `removeFromRegister` and `clearFileRegister`

**Comment by MAXakaWIZARD** [comment]
@wysow Any news when this PR can be merged?

**Pull request merged**